runtime/src/main/java/org/corfudb/runtime/collections/CorfuStreamEntry.java

```diff
@@ -19,25 +18,31 @@
  * @param <M> - type of the protobuf metadata schema defined by table creation.
  */
 @Slf4j
-@EqualsAndHashCode
-public class CorfuStreamEntry<K extends Message, V extends Message, M extends Message> {
-    /**
-     * Key of the UFO stream entry
-     */
-    @Getter
-    private final K key;
+public class CorfuStreamEntry<K extends Message, V extends Message, M extends Message> extends
+        CorfuStoreEntry<K, V, M> {
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        if (!super.equals(o)) {
+            return false;
+        }
 
-    /**
-     * Value of the UFO stream entry
-     */
-    @Getter
-    private final V payload;
+        CorfuStreamEntry<K, V, M> that = (CorfuStreamEntry<K, V, M>) o;
 
-    /**
-     * Metadata (ManagedResource) of the UFO stream entry
-     */
-    @Getter
-    private final M metadata;
+        return getOperation() == that.getOperation();
+    }
+
+    @Override
+    public int hashCode() {
+        int result = super.hashCode();
+        result = 31 * result + getOperation().hashCode();
```

Comment:
how can we be certain that this is the right way to get the hashcode for 'operation'?

Reply:
This is autogenerated code from the IDE